server/form_validation/__init__.py

```diff
@@ -0,0 +1,104 @@
+from jsonschema import Draft7Validator, validators, ValidationError
+from jsonschema.exceptions import best_match
+from copy import deepcopy
+
+def extend_with_default(validator_class=Draft7Validator):
+    """
+    Extends a validator class to add defaults before validation.
+    From: https://github.com/Julian/jsonschema/blob/master/docs/faq.rst
+    """
+    validate_props = validator_class.VALIDATORS["properties"]
+    validate_array = validator_class.VALIDATORS["items"]
+    validate_oneOf = validator_class.VALIDATORS["oneOf"] # must use draft 4+
+
+    def set_defaults(validator, properties, instance, schema):
+        """ Set defaults within a "properties" context """
+        if not validator.is_type(instance, "object"):
+            return
+        for prop, subschema in properties.items():
+            if instance.get(prop) == None:
+                if "default" in subschema:
+                    default_val = subschema["default"]
+                elif subschema.get('type') == 'array':
```

Comment:
Use one kind of quote consistently. I don't care which, but it seems that double quotes are more common.